src/api/app/models/package.rb

```diff
@@ -1134,10 +1155,14 @@ def save_file(opt = {})
     content = opt[:file] if opt[:file]
     logger.debug "storing file: filename: #{opt[:filename]}, comment: #{opt[:comment]}"
 
+    Package.verify_file!(self, opt[:filename], opt[:file])
+    unless User.current.can_modify_package?(self)
```

Comment:
I agree with you guys... it should be checked in the controller side using a policy. But have in mind that my target is "Get rid of FrontendCompat in webui/driver_update_controller.rb" nothing more...

That change was suggested by Adrian because it wasn't checked there before and it should be. And if you see the model it's done the same way several times...

For me is ok right now because I'm not taking care of the whole model (I know that is bad done right now). I created a new Trello card to refactor the whole Package model and it should involve the creation of the policy.